File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/displayhints/DisplayHintsAccountOnFile.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct.displayhints;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import com.onlinepayments.sdk.client.android.model.paymentproduct.AccountOnFileDisplay;

/**
 * POJO that represents an DisplayHintsAccountOnFile object.
 */
public class DisplayHintsAccountOnFile implements Serializable {

	private static final long serialVersionUID = 3446099654728722104L;

	private List<AccountOnFileDisplay> labelTemplate = new ArrayList<>();

	public List<AccountOnFileDisplay> getLabelTemplate(){
		return labelTemplate;
	}

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public DisplayHintsAccountOnFile() {}
}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/displayhints/DisplayHintsAccountOnFile.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct.displayhints

import com.onlinepayments.sdk.client.android.model.paymentproduct.AccountOnFileDisplay
import java.io.Serializable

/**
 * Data class that represents an DisplayHintsAccountOnFile object.
 */
data class DisplayHintsAccountOnFile(
    val labelTemplate: MutableList<AccountOnFileDisplay?> = mutableListOf()
) : Serializable {

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = 3446099654728722104L
    }
}
